src/types.rs

```diff
@@ -245,6 +245,71 @@ pub enum SignedTransaction {
     Eip4844(SignedEip4844Transaction),
 }
 
+impl SignedTransaction {
```

Comment:
Hmm, this feels like a code smell that indicates the types should be refactored somehow. Let me think about it a bit, I may have some nice alternatives to this.